### PR TITLE
Hash and verify API Secret

### LINKED ISSUE #3: Is it okay to never store the API Secret, and to use it to encrypt the TOTP Keys?
We considered using the API Secret (that we generate and provide to the consumer of the API) as the AES key for encrypting the TOTP Keys. Then, since we would not store the API Secret, we would not be able to expose the TOTP Keys because we couldn't even decrypt a consumer's TOTP Keys until they call us with the correct API Secret.

However, if we do that then we have no way to know whether they are in fact using the API Secret we gave them. We would have to consider their API Key as sufficient to confirm that they are who they say they are, and they could pass us "abc" as their API Secret and we might not know the difference.

Is that sufficient reason for us to store _something_ related to their API Secret? We could treat it as a password, and store a salted, stretched hash of it, then verify that hash before using the given API Secret to encrypt/decrypt their TOTP Key(s).

Thoughts?

### PR DESCRIPTION
**Pull request by forevermatt**
Closes #3

### CODE REVIEW

File: features/step-definitions/all-steps.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  let apiSecret;
+  let apiSecret2;
+  let apiSecretValidationResult;
+  let decryptedMessage;
+  let encryptedMessage;
+  let hashedApiSecret;
+  let hashedApiSecretCheck;
+  let plainTextMessage;
+  
+  Given('I do not have an API Key record', function () {
```
Comment (wcjr) [inline features/step-definitions/all-steps.js]: to be consistent, you capitalized NOT on 24...

File: models/api-key.js [hunk 2]
```diff
@@ -114,6 +161,29 @@ const isAlreadyActivated = (apiKeyRecord) => {
 };
 module.exports.isAlreadyActivated = isAlreadyActivated;
 
+const isValidApiSecret = (apiKeyRecord, apiSecret = '') => {
+  if (!apiKeyRecord) {
```
Comment (wcjr) [inline models/api-key.js]: might consider moving these timing-protection lines into compare for cohesion.

File: features/step-definitions/all-steps.js (hunk 1 above)
Comment (forevermatt) [reply features/step-definitions/all-steps.js]: Thanks, fixed in 2d19752

File: models/api-key.js (hunk 2 above)
Comment (forevermatt) [reply models/api-key.js]: Sounds good. I may do that as a separate PR just to avoid changing this PR too much after approvals.

**Pull request merged**